tests/queries/test_query.py

```diff
@@ -165,6 +165,18 @@ def test_equality(self):
             Author.objects.filter(Q(item__name='foo')).query,
         )
 
+    def test_equality_with_negated_nullable(self):
+        # nullable field
+        self.assertEqual(
+            str(Item.objects.exclude(modified__year='2021').values('id').query),
+            str(Item.objects.annotate(year=ExtractYear('modified')).exclude(year='2021').values('id').query),
+        )
+        # not nullable field
+        self.assertEqual(
+            str(Item.objects.exclude(created__year='2021').values('id').query),
+            str(Item.objects.annotate(year=ExtractYear('created')).exclude(year='2021').values('id').query),
+        )
```

Comment:
Comparing raw queries is the last resort, in this case we should be able to create a queryset that doesn't return the correct results without a fix:
```python
self.assertCountEqual(
    queryset,
    [....],
)
```